pennylane/compiler/python_compiler/quantum_dialect.py

```diff
@@ -203,6 +206,53 @@ class CustomOp(IRDLOperation):
 
     out_ctrl_qubits = var_result_def(BaseAttr(QubitType))
 
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        *,
+        in_qubits: (
+            SSAValue[QubitType] | Operation | Sequence[SSAValue[QubitType]] | Sequence[Operation]
+        ),
+        gate_name: str | StringAttr,
+        params: SSAValue[Float64Type] | Sequence[SSAValue[Float64Type]] = (),
+        in_ctrl_qubits: (
+            SSAValue[QubitType] | Operation | Sequence[SSAValue[QubitType]] | Sequence[Operation]
+        ) = (),
+        in_ctrl_values: (
+            SSAValue[IntegerType]
+            | Operation
+            | Sequence[SSAValue[IntegerType]]
+            | Sequence[Operation]
+        ) = (),
```

Comment:
I think it is always best to have optional values be defaulted to None

```suggestion
            | Sequence[Operation]
            | None
        = None
```

Same for the ones above.